e2e/terraform/shared/config/provision-server.sh

```diff
@@ -3,39 +3,33 @@
 set -o errexit
 set -o nounset
 
-CLOUD="$1"
-SERVER_COUNT="$2"
-NOMAD_CONFIG="$3"
```

Comment:
Note for reviewers: the indexed servers config here was removed because all 3 versions were always identical. There'll be support in the `custom` profile for this use case but it's not necessary here.